plugins/inputs/snmp_trap/snmp_trap.go

```diff
@@ -0,0 +1,144 @@
+package snmp_trap
+
+import (
+	"fmt"
+	"log"
+	"net"
+	"strconv"
+	"sync"
+	"time"
+
+	"github.com/influxdata/telegraf"
+	"github.com/influxdata/telegraf/plugins/inputs"
+	"github.com/influxdata/telegraf/plugins/inputs/snmp"
+
+	"github.com/soniah/gosnmp"
+)
+
+type SnmpTrap struct {
+	Port uint16 `toml:"port"`
+
+	acc      telegraf.Accumulator
+	listener *gosnmp.TrapListener
+	wg       sync.WaitGroup
+	timeFunc func() time.Time
+
+	makeHandlerWrapper func(func(packet *gosnmp.SnmpPacket, addr *net.UDPAddr)) func(packet *gosnmp.SnmpPacket, addr *net.UDPAddr)
+	Errch              chan error
+}
+
+var sampleConfig = `
+  ## Port to listen on.  Default 162
+  #port = 162
```

Comment:
This would require a change to gosnmp.  "udp" is hard coded in (t *TrapListener) Listen().

https://github.com/soniah/gosnmp/blob/master/trap.go#L154